### DISCUSSION TITLE
use sdkman with internal artifactory maven mirror

### DISCUSSION THREAD

**Discussion by zisuu** [sdkman/sdkman-cli#1322] [Q&A]
Hello

I love sdkman for my private pet projects. However, in my company, I can not use sdkman because we are forced to use our internal Maven and Java mirrors to download specific Java or Maven versions. Is there a way to tell sdkman where to look at our Artifactory to pull Java or Maven versions?

For Maven, we have a custom settings.xml file in our .m2 home directory, which points maven to our Artifactory:
```xml
  <mirrors>
    <mirror>
      <mirrorOf>*</mirrorOf>
      <name>company</name>
      <url>https://company.internal/artifactory/company</url>
      <id>company</id>
    </mirror>
  </mirrors>
```

is there a way to use this also with sdkman? thanks a lot!

**Comment by marc0der** [comment]
Unfortunately, this is not possible as our database references the actual resources on the web. When installing an artefact with SDKMAN, you aren't downloading it from us but rather from the actual location on the web where each vendor hosts their SDKs.

The above snippet you posted is a Maven config, but we don't use Maven to fulfil downloads. All downloads occur by first querying our database for the actual resource location, then we curl it down from that location to your machine and install it.

The only way you could get around this is by spinning up your own version of the SDKMAN backend and populating the MongoDB store with your own data, each version pointing to a location you determine. This will be a lot of effort on your behalf and not something we would be able to support.
